src/etl/v2g/functional_predictions/vep.py

```diff
@@ -0,0 +1,201 @@
+"""The variant annotation dataset contains information about the impact of a variant on a transcript or protein. These can be mapped to genes allowing us to establish significant relationships between variants and genes."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import pyspark.sql.functions as f
+
+from etl.common.spark_helpers import get_record_with_maximum_value
+
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame
+
+    from etl.common.ETLSession import ETLSession
+
+
+def main(
+    etl: ETLSession,
+    variant_index_path: str,
+    variant_annotation_path: str,
+    variant_consequence_lut_path: str,
+) -> tuple[DataFrame, ...]:
+    """Extracts variant to gene assignments for the variants included in the index and the features predicted by VEP.
+
+    Args:
+        etl (ETLSession): ETL session,
+        variant_index_path (str): The path to the OTG variant index
+        variant_annotation_path (str): The path to the variant annotation file
+        variant_consequence_lut_path (str): The path to the LUT between the functional consequences and their assigned V2G score
+
+    Returns:
+        DataFrame: variant to gene assignments from VEP
+    """
+    # function that takes the index, gets the vep data and parses it
+    etl.logger.info("Loading variant index")
+    variant_consequence_lut = read_consequence_lut(etl, variant_consequence_lut_path)
+    va_with_vep = (
+        etl.spark.read.parquet(variant_annotation_path)
+        # exploding the array already removes record without VEP annotation
+        .select(
+            "id", f.explode("vep.transcriptConsequences").alias("transcriptConsequence")
+        )
+    )
+    vi = etl.spark.read.parquet(variant_index_path).select("id")
+    annotated_variants = (
+        va_with_vep.join(vi, on="id", how="inner").coalesce(20).persist()
```

Comment:
`coalesce(20)`?

Reply:
`va_with_vep` has  440M rows in 400 partitions. `vi` has 6M rows in 200 partitions.
The intersection of both is a much smaller dataset with 4M rows that I don't need to have distributed in a lot of partitions, that is why I am downgrading to 20. I use 20 because I read that the number of partitions should be a multiple of the default number of nodes (200).